vcd/resource_vcd_network_isolated_v2_nsxt_test.go

```diff
@@ -157,3 +159,262 @@ resource "vcd_network_isolated_v2" "net1" {
+
+	params["FuncName"] = t.Name() + "-step4"
+	configText4 := templateFill(testAccVcdNetworkIsolatedV2NsxtMigrationStep4, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 4: %s", configText4)
+
+	params["FuncName"] = t.Name() + "-step5"
+	configText5 := templateFill(testAccVcdNetworkIsolatedV2NsxtMigrationStep3, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 5: %s", configText5)
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	cachedId := &testCachedFieldValue{}
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: testAccProviders,
+		PreCheck:          func() { testAccPreCheck(t) },
+		CheckDestroy:      testAccCheckOpenApiVcdNetworkDestroy(testConfig.Nsxt.Vdc, t.Name()),
+		Steps: []resource.TestStep{
+			{ // step 1 - setup prerequisites
+				Config: configTextPre,
+			},
+			{
+				Config: configText2,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					cachedId.cacheTestResourceFieldValue("vcd_network_isolated_v2.net1", "id"),
+					resource.TestCheckResourceAttrSet("vcd_network_isolated_v2.net1", "id"),
+					resource.TestMatchResourceAttr("vcd_network_isolated_v2.net1", "owner_id", regexp.MustCompile(`^urn:vcloud:vdc:`)),
+					resource.TestCheckResourceAttr("vcd_network_isolated_v2.net1", "is_shared", "false"),
+				),
+			},
+			{
+				Config: configText3,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					cachedId.testCheckCachedResourceFieldValue("vcd_network_isolated_v2.net1", "id"),
+					resource.TestCheckResourceAttrSet("vcd_network_isolated_v2.net1", "id"),
+					resource.TestMatchResourceAttr("vcd_network_isolated_v2.net1", "owner_id", regexp.MustCompile(`^urn:vcloud:vdc:`)),
+					resource.TestCheckResourceAttr("vcd_network_isolated_v2.net1", "is_shared", "false"),
+				),
+			},
+			{
+				Config: configText4,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					cachedId.testCheckCachedResourceFieldValue("vcd_network_isolated_v2.net1", "id"),
+					resource.TestCheckResourceAttrSet("vcd_network_isolated_v2.net1", "id"),
+					resource.TestMatchResourceAttr("vcd_network_isolated_v2.net1", "owner_id", regexp.MustCompile(`^urn:vcloud:vdcGroup:`)),
+					resource.TestCheckResourceAttr("vcd_network_isolated_v2.net1", "is_shared", "true"),
+				),
+			},
+			{
+				Config: configText5,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					cachedId.testCheckCachedResourceFieldValue("vcd_network_isolated_v2.net1", "id"),
+					resource.TestCheckResourceAttrSet("vcd_network_isolated_v2.net1", "id"),
+					resource.TestMatchResourceAttr("vcd_network_isolated_v2.net1", "owner_id", regexp.MustCompile(`^urn:vcloud:vdc:`)),
+					resource.TestCheckResourceAttr("vcd_network_isolated_v2.net1", "is_shared", "false"),
+				),
+			},
+		},
+	})
+	postTestChecks(t)
+}
+
+const testAccVcdNetworkIsolatedV2NsxtMigrationStep2 = testAccVcdVdcGroupNew + `
+resource "vcd_network_isolated_v2" "net1" {
+  org  = "{{.Org}}"
+  vdc  = vcd_org_vdc.newVdc.0.name
```

Comment:
I probably don't see it, but do we have a test which checks when `vdc` is defined at provider level?

Reply:
Yep, good ask. There was an initial reason why I did so - our provider by default hardcodes NSX-V VDC name in provider. I did a few manual tests with inherited VDC field, but that is not good enough. I will think about this now to address it.

Reply:
After prolonged investigation I found a method to override `vdc` field configuration in `provider` using new function `overrideDefaultVdcForTest` in particular tests. As a result there are 3 new tests in commit 1fe13f4 which cover inheriting `vdc` from `provider` and migrating the setting to resources (also checks data sources):
* `TestAccVcdNetworkIsolatedV2InheritedVdc` 
* `TestAccVcdNetworkRoutedV2InheritedVdc` (this also covers NSX-T Edge Gateway creation as routed network requires it)
* `TestAccVcdNetworkImportedV2InheritedVdc`

Good news is that no new bugs were found and this commit only affects tests.